### PR TITLE
feat(v2): upgrade embedded-metrics to v4

### PR DESCRIPTION
**Pull request by jeromevdl**
**Issue #, if available:** #1041

## Description of changes:

Upgrade to latest version of embedded-metrics

**Checklist**

 

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [x] Update tests
* [ ] Update docs ==> not yet
* [x] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

Checked exception instead of unchecked exception. I feel the new dev experience is shitty with these exceptions. Is there anything we can do?

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jeromevdl** [comment]
@scottgerring I'm not happy with this, have a look at the examples, the experience is really bad now with these checked exceptions. We encapsulate some of them, should we go further? Review the module completely?

**Comment by scottgerring** [comment]
@jeromevdl I linked this to #1041 - do we cover off the whole thing here (high-res metrics), or just the upgrade?

**Comment by jeromevdl** [comment]
I didn't check high-res, it's just an upgrade (don't know if high res is handled directly or if we need to do something).

**Comment by scottgerring** [comment]
> I didn't check high-res, it's just an upgrade (don't know if high res is handled directly or if we need to do something).

Should we look at high-res as part of this, or unlink it from the underlying issue and deal with that separately? It's one of the things on the V2 roadmap.

**Comment by jeromevdl** [comment]
> > I didn't check high-res, it's just an upgrade (don't know if high res is handled directly or if we need to do something).
> 
> Should we look at high-res as part of this, or unlink it from the underlying issue and deal with that separately? It's one of the things on the V2 roadmap.

high res is provided with the new version, we have nothing to do on our side.

File: examples/powertools-examples-core-utilities/cdk/app/src/main/java/helloworld/App.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
  * Handler for requests to Lambda function.
  */
 public class App implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
-    private final static Logger log = LogManager.getLogger(App.class);
+    private static final Logger log = LogManager.getLogger(App.class);
```
Comment (scottgerring) [inline examples/powertools-examples-core-utilities/cdk/app/src/main/java/helloworld/App.java:47]: What's up with all the whitespace changes?

**Review by scottgerring — COMMENTED** [review]
Looks good and straightforward. Still not sure about the exception thing though ... I feel like this is the best of the bad options, but perhaps it is worth checking with the library why they've done it this way

**Comment by scottgerring** [comment]
https://github.com/awslabs/aws-embedded-metrics-java/releases/tag/v4.1.2 🥳

**Comment by scottgerring** [comment]
We need to redo this from the released version of the library without the checked exceptions we have extensively worked around here

**Pull request closed without merging**